src/analysis/scripts/download_census_lodes.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+
+"""
+A script to cache Census LODES data on S3 to avoid downloading the same file repeatedly.
+
+Given a local directory, a state, and an S3 bucket, it
+- Checks first if the LODES file is already in the local directory, and does nothing if so
+- Tries to download the LODES file for the state and data type from the S3 bucket
+- If it's not there, tries to download the file from the Census site and upload it to the S3 bucket
+- It it's not found, tries to download the file from the previous year from S3 and then from Census
+
+If run without a state defined, it will run for all states, deleting each local download as it goes
+(so as to not overwhelm the local disk space). This option exists to build the S3 cache.
+
+If run without a data type defined, it will run for both data types.
+(This option is only expected to be run in conjunction with no set state, to build the S3 cache.)
+"""
+import argparse
+import gzip
+import logging
+import os
+import shutil
+import sys
+
+import boto3
+from botocore.exceptions import ClientError
+from botocore.exceptions import ProfileNotFound
+import requests
+import us
+
+logging.basicConfig(
+    stream=sys.stderr,
+    format='{} %(asctime)s %(levelname)-8s %(message)s'.format(os.path.basename(__file__)),
+    datefmt='%Y-%m-%d %H:%M:%S'
+)
+logger = logging.getLogger(__file__)
+
+LODES_DATA_TYPES = ('main', 'aux')
+# fall back to an older year if the most recent in use is not available
+LODES_MOST_RECENT_YEAR = '2017'
+LODES_FALLBACK_YEAR = '2016'
+LODES_URL = 'http://lehd.ces.census.gov/data/lodes/LODES7'
+S3_LODES_DIRECTORY = 'data/'
```

Comment:
I think this trailing slash is causing the S3 keys to come out like `data//hi_od_main_JT00_2017.csv.gz`.